RCAIDE/Library/Methods/Thermal_Management/Heat_Exchangers/Cross_Flow_Heat_Exchanger/design_cross_flow_heat_exchanger.py

```diff
@@ -0,0 +1,164 @@
+from RCAIDE.Library.Methods.Thermal_Management.Heat_Exchangers.Cross_Flow_Heat_Exchanger.cross_flow_heat_exchanger_sizing_setup import cross_flow_heat_exchanger_sizing_setup 
+from RCAIDE.Library.Methods.Thermal_Management.Heat_Exchangers.Cross_Flow_Heat_Exchanger.cross_flow_heat_exchanger_geometry_setup import cross_flow_heat_exchanger_geometry_setup
+from RCAIDE.Framework.Optimization.Common             import Nexus
+from RCAIDE.Framework.Optimization.Packages.scipy     import scipy_setup
+
+# Python package imports   
+import numpy as np  
+import time 
+
+# ----------------------------------------------------------------------
+#  Rotor Design
+# ----------------------------------------------------------------------
+## @ingroup Methods-Thermal_Management-Batteries
+def design_cross_flow_heat_exchanger(HEX,HAS,battery, single_side_contact=True, dry_mass=True,
+                                       solver_name= 'SLSQP',iterations = 200,solver_sense_step = 1E-2,
+                                       solver_tolerance = 1E-1,print_iterations = False):
+    
+    """ Optimizes heat exchanger geometric properties input parameters to minimize either design power and mass. 
+        This scrip adopts RCAIDE's native optimization style where the objective function is expressed 
+        as a sizing function, considering both power and mass.
+          
+          Assumptions: 
+            
+        
+          Source:
+           Shah RK, Sekulić DP. Fundamentals of Heat Exchanger Design. John Wiley & Sons; 2003. 
+    """    
+    
+    
+    # start optimization 
+    ti                   = time.time()  
+    optimization_problem = crossflow_heat_exchanger_design_problem_setup(HEX,HAS,print_iterations)
+    output               = scipy_setup.SciPy_Solve(optimization_problem,solver=solver_name, iter = iterations , sense_step = solver_sense_step,tolerance = solver_tolerance)  
+    
+    tf                   = time.time()
+    elapsed_time         = round((tf-ti)/60,2)
+    print('HEX sizing optimization Simulation Time: ' + str(elapsed_time) + ' mins')    
+ 
+    HEX_opt = optimization_problem.hex_configurations.optimized.networks.electric.busses.bus.batteries.lithium_ion_nmc.thermal_management_system.heat_exchanger_system 
+    HEX.design_air_inlet_pressure     = output[0]
+    HEX.design_coolant_inlet_pressure = output[1]
+    HEX.design_air_mass_flow_rate     = output[2]
+    HEX.design_coolant_mass_flow_rate = output[3]
+    HEX.stack_length                  = HEX_opt.stack_length
+    HEX.stack_width                   = HEX_opt.stack_width
+    HEX.stack_height                  = HEX_opt.stack_height
+    HEX.design_air_frontal_area       = HEX_opt.air_frontal_area
+    HEX.design_air_pressure_diff      = HEX_opt.pressure_diff_air   
+    return HEX
+
+## @ingroup Methods-Thermal_Management-Batteries-Sizing
+def crossflow_heat_exchanger_design_problem_setup(HEX,HAS,print_iterations): 
+    """
+    Optimizer function
```

Comment:
Keep the function header consistent with RCAIDE

Reply:
Done:
included same changes for wavy channel as well